src/txrequest.h

```diff
@@ -208,6 +208,15 @@ class TxRequestTracker {
 
     /** Access to the internal PriorityComputer (for testing) */
     const PriorityComputer& GetPriorityComputer() const;
+
+    /** Run internal consistency check (testing only). */
```

Comment:
Wouldn't hurt to make this block not publicly callable somehow.

Just an idea in case it's not crazy: https://stackoverflow.com/a/23267346

Reply:
I'd prefer to move these into a friend `TxRequestTrackerTester` class (I thought I'd left a review comment suggesting that, but I appear to have lost it).

Reply:
@jnewbery How do you suggest to do that in combination with the impl pattern?

Reply:
Hmmm yes, very good question! I wonder if something along the lines of https://stackoverflow.com/a/34054983/933705 might work (but then again, it might not be worth it just to avoid putting these three test helper functions in the public interface).

Reply:
I think it's not unreasonable to have sanity checking functions for involved data structures in code that actually deals with that data structure. It could possibly be called at runtime using something something similar to `-checkmempool` too if we wanted.

Reply:
I agree. That seems reasonable.